fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/MetaDataProtoEditor.java

```diff
@@ -174,16 +363,19 @@ public static void addNestedRecordType(@Nonnull RecordMetaDataProto.MetaData.Bui
      * @param metaDataBuilder the meta-data builder
      * @param recordType the record type to be deprecated
      */
-    @Nonnull
     public static void deprecateRecordType(@Nonnull RecordMetaDataProto.MetaData.Builder metaDataBuilder, @Nonnull String recordType) {
-        DescriptorProtos.DescriptorProto.Builder unionBuilder = fetchUnionBuilder(metaDataBuilder.getRecordsBuilder());
+        final DescriptorProtos.FileDescriptorProto.Builder fileBuilder = metaDataBuilder.getRecordsBuilder();
+        DescriptorProtos.DescriptorProto.Builder unionBuilder = fetchUnionBuilder(fileBuilder);
         if (unionBuilder.getName().equals(recordType)) {
             throw new MetaDataException("Cannot deprecate the union");
         }
         // deprecate all fields of type recordType from the union.
         boolean found = false;
         for (DescriptorProtos.FieldDescriptorProto.Builder fieldBuilder : unionBuilder.getFieldBuilderList()) {
-            if (fieldBuilder.getTypeName().equals(recordType)) {
+            final FieldTypeMatch fieldTypeMatch = fieldIsType(fileBuilder, unionBuilder, fieldBuilder, recordType);
+            if (fieldTypeMatch.isPossibleMatch()) {
```

Comment:
I now see why `MATCHES.possibleMatch = false`. But I think it may be more intuitive, if it was `true` but instead this `if` was the other way around:
``` 
if (FieldTypeMatch.MATCHES.equals(fieldTypeMatch)) {
...
} else if (fieldTypeMatch.isPossibleMatch()) {
...
}
```

You could also just rename it to something more explicit (e.g., `possibleButNotDefinite` or something) .

Reply:
sure, the name looks great. thanks.